### ISSUE TITLE
armasm64.exe on Windows doesn't support -D flags

### ISSUE THREAD

**Issue by tleasor** [alexcrichton/cc-rs#516]
The aarch64 assembler on Windows, `armasm64.exe` doesn't support -D flags set using .define(). This causes a build failure to occur when targeting aarch64-pc-windows-msvc using `cc`. I'm happy to commit a fix using the equivalent -predefine flag (http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.kui0100a/armasm_cacbjgdb.htm) for this target if that's an acceptable solution.

**Comment by alexcrichton** [comment]
Sounds good to me! A PR would be most welcome :)

**Comment by Stammark** [comment]
I also ran into this :) (but for my purposes I just disabled cc-rs from passing the -D option because my assembler snippet didn't care)

I'd be wary of calling --predefine an equivalent flag, though. (https://developer.arm.com/docs/101754/0614/armasm-legacy-assembler-reference/armasm-command-line-options/-predefine-directive : "--predefine defines a global variable, whereas -Dname defines a macro that the preprocessor expands" (think #define , #ifdef etc.), but I don't know enough about the legacy assembler syntax to know the direct impact in data allocation or symbol creation.

This link suggests "Although you can use predefined global variables in combination with assembly control directives, for example IF and ELSE to control conditional assembly, they are not intended to provide the same functionality as the C preprocessor in armasm. If you require this functionality, Arm recommends you use the compiler to pre-process your assembly code."
Can that be done with MSVC, though.... I'm not sure!

**Comment by tleasor** [comment]
@Stammark Thanks for the link, that helps to clarify things. When I ran into this issue myself (trying to build a native rustc) the [assembly file in question](https://github.com/rust-lang/stacker/blob/master/psm/src/arch/aarch64_armasm.asm) didn't make use of the -D flags so I used the same workaround you described.

It looks like MSVC might support that since it's not listed in the [differences of their armasm version](https://docs.microsoft.com/en-us/cpp/assembler/arm/arm-assembler-directives?view=vs-2019). However, given that the flags aren't equivalent and assembly authors targeting Windows ARM64 would most likely be aware of MSVC limitations, the best solution would seem to be to skip defining any -D flags for this target. To that end, I've opened a simple PR with a fix in #518.
